cirq-core/cirq/sim/simulator.py

```diff
@@ -235,6 +236,64 @@ def compute_amplitudes_sweep_iter(
         """
         raise NotImplementedError()
 
+    def sample_from_amplitudes(
+        self,
+        circuit: 'cirq.AbstractCircuit',
+        param_resolver: 'cirq.ParamResolver',
+        repetitions: int = 1,
+        qubit_order: 'cirq.QubitOrderOrList' = ops.QubitOrder.DEFAULT,
+    ) -> Dict[int, int]:
```

Comment:
Might want to return a `cirq.Result` here.

Reply:
Unfortunately, this can't be reshaped into a `Result`. It's a histogram of samples from the final state of the circuit, which does not have the measurement keys required for constructing a `Result`.